File: Playwright/WEB_TEST/Client/elements/PowerGridMetrics/DataPanels/controlPanelElements.js
import PowerGridMetricsLocators from "../../../locators/PowerGridMetrics/powerGridMetricsLocators";
import TestHelpers from "../../../../Common/modules/testHelpers";
import BaseControlPanelElements from "../../common/DataPanels/baseControlPanelElements";

export class ControlPanelElements extends BaseControlPanelElements {

    constructor(page, parentLocator = null) { 
        super(page, PowerGridMetricsLocators.controlPanelLocators)
        this.parentLocator = parentLocator ? page.locator(parentLocator) : page;
    }

    showReportTgButton() {
        return TestHelpers.getLinkedLocator(this.controlPanel(), PowerGridMetricsLocators.controlPanelLocators.showReportTgButton.locator)
    }

    getCurrentDataButton() {
        return TestHelpers.getLinkedLocator(this.controlPanel(), PowerGridMetricsLocators.controlPanelLocators.currentDataButton.locator)
    }

    showReportButton() {
        return TestHelpers.getLinkedLocator(this.controlPanel(), PowerGridMetricsLocators.controlPanelLocators.showReportButton.locator)
    }

    gridMetricsParamsButton() {
        return TestHelpers.getLinkedLocator(this.controlPanel(), PowerGridMetricsLocators.controlPanelLocators.gridMetricsParamsButton.locator)
    }

    exportFormatOption(format) {
        return this.page.locator(PowerGridMetricsLocators.controlPanelLocators.exportFormatOption.locator(format))
    }

}

export default ControlPanelElements

File: Confluence exporter/grafana/elements/controlPanelElements.py
from src.modules.grafana.locators.controlPanelLocators import ControlPanelLocators


class ControlPanelElements:
    def __init__(self, page):
        self.page = page
        self.control_panel_locators = ControlPanelLocators()

    def timepicker(self):
        return self.page.locator(self.control_panel_locators.TIMEPICKER)

    def date_from_input(self):
        return self.page.locator(self.control_panel_locators.DATE_FROM_INPUT)

    def date_to_input(self):
        return self.page.locator(self.control_panel_locators.DATE_TO_INPUT)

    def apply_date_button(self):
        return self.page.locator(self.control_panel_locators.APPLY_DATE_BUTTON)

    def save_dashboard_button(self):
        return self.page.locator(self.control_panel_locators.SAVE_DASHBOARD_BUTTON)
